homeassistant/components/samsungtv/config_flow.py

```diff
@@ -182,3 +184,7 @@ def _try_connect(self):
         return self.async_show_form(
             step_id="confirm", description_placeholders={"model": self._model}
         )
+
+    async def async_step_reauth(self, user_input=None):
+        """Handle configuration by re-auth."""
+        return await self.async_step_confirm(user_input)
```

Comment:
Yes, if we're creating a new entry with the same unique_id, the new entry will replace the old one.

We could add logic in the config flow that just updates the existing entry and aborts instead of creating a new entry.

`self._abort_if_unique_id_configured` now supports updates.

Reply:
Theoretically, for the re-auth we don't need any new/changed entry. It's just the popup on the TV which needs to be acknowleged.